Pokedex/View Controllers/MainTableViewController.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  MainTableViewController.swift
+//  Pokedex
+//
+//  Created by krikaz on 6/9/20.
+//  Copyright © 2020 thewire. All rights reserved.
+//
+
+import UIKit
+
+class MainTableViewController: UITableViewController {
+
+    private let baseURL = URL(string: "https://pokeapi.co/api/v2/")!
+    private let cellReuseID = "PokemonCell"
+    private var allPokemons: AllPokemons? {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+            
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = "Pokemons"
+        self.getAllPokemons()
+    }
+    
+    private func getAllPokemons() {
+        var url = baseURL
+        url.appendPathComponent("pokemon")
+        var urlComp = URLComponents(url: url, resolvingAgainstBaseURL: false)
+        urlComp?.queryItems = [URLQueryItem(name: "limit", value: "100"), URLQueryItem(name: "offset", value: "0")]
+        
+        
+        guard let url1 = urlComp?.url else { return }
+//        print(url1)
+        var request = URLRequest(url: url1)
+        request.httpMethod = HTTPMethod.get.rawValue
+        
+        let task = URLSession.shared.dataTask(with: request) { (data, _, error) in
+            if let error = error {
+                NSLog("Error: \(error)")
+                return
+            }
+            
+            guard let data = data else { return }
+            do {
+                self.allPokemons = try JSONDecoder().decode(AllPokemons.self, from: data)
+//                print(self.allPokemons?.results)
+//                print(self.allPokemons?.next)
+//                print(self.allPokemons?.previous)
+            } catch {
+                NSLog("Error: \(error)")
+            }
+        }
+        task.resume()
+    }
```

Comment:
Also network logic. If you have a class for this logic, you can reuse it in multiple places. In reality, it wouldn't be in the ViewController but for learning purposes, you can leave it here.